parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -438,8 +438,28 @@ configGPUDrivers() {
     retrycmd_if_failure 120 5 25 $GPU_DEST/bin/nvidia-smi || exit $ERR_GPU_DRIVERS_START_FAIL
     retrycmd_if_failure 120 5 25 ldconfig || exit $ERR_GPU_DRIVERS_START_FAIL
 }
+
+validateGPUDrivers() {
+    retrycmd_if_failure 24 5 25 nvidia-modprobe -u -c0 && echo "gpu driver loaded" || configGPUDrivers || exit $ERR_GPU_DRIVERS_START_FAIL
+    SMI_RESULT=$(retrycmd_if_failure 24 5 25 $GPU_DEST/bin/nvidia-smi)
+    SMI_STATUS=$?
+    if [[ $SMI_STATUS != 0 ]]; then
+        if [[ $SMI_RESULT == *"infoROM is corrupted"* ]]; then
+            exit $ERR_GPU_INFO_ROM_CORRUPTED
+        else
+            exit $ERR_GPU_DRIVERS_START_FAIL
+        fi
+    else
+        echo "gpu driver working fine"
+    fi
+}
+
 ensureGPUDrivers() {
-    configGPUDrivers
+    if [[ "${CONFIG_GPU_DRIVER_IF_NEEDED}" = true ]]; then
+        configGPUDrivers
+    else
+        validateGPUDrivers
```

Comment:
On a second thought - do you think we should always do the validation for vhd1/2/3? Of course we need some small refactoring such as  skipping L437/L438 in configGPU, revisit the retry counts n L443/L444, and also some manual test for VHD1.

feel free to ignore ifyou don't want to get distracted

Reply:
The benefit is we also get the inforRom corrupted signal, maybe another PR to address it separately

Reply:
prefer another PR. I think we do want some refactoring to group validation together